mlir/include/mlir/Dialect/MIOpen/Tuning/GridwiseGemmParams.h

```diff
@@ -781,42 +654,60 @@ class PopulateParamsXDL : public PopulateParamsBase {
     return success();
   }
 
-  LogicalResult isValidXDLOPSGemm(InitParamsXDL *param, int64_t blockSize) {
+  LogicalResult isValidblockwisegemmxdlops(InitParamsXDL &param,
+                                           int64_t blockSize) {
     // TBD: support fp16/bf16
-    const auto gemmKPackedPerBlock = param->gemmKPerBlock;
 
-    // unsupported xdlops-gemm
-    if (param->gemmMPerWave == 16 && param->gemmNPerWave == 32)
-      return failure();
-    if (param->gemmMPerWave == 32 && param->gemmNPerWave == 16)
-      return failure();
-    if (param->gemmMPerWave == 8 && param->gemmNPerWave != 64)
-      return failure();
-    if (param->gemmMPerWave == 4 && param->gemmNPerWave != 64)
-      return failure();
-    if (param->gemmMPerWave == 32 && param->gemmNPerWave == 32 &&
-        gemmKPackedPerBlock % 2 != 0)
-      return failure();
-    if (param->gemmMPerWave == 16 && param->gemmNPerWave == 16 &&
-        gemmKPackedPerBlock % 4 != 0)
+    std::vector<std::tuple<int, int, int>> validWaveGemmSize = {
+        // std::make_tuple(128, 128, 1),
```

Comment:
@jerryyin please remove unused ones.

Reply:
This is ported from [here](https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/src/solver/implicitgemm_util.hpp#L544-L563). I try not confuse myself in case we want to make updates to this function later therefore I copied it as-is.